soes/esc_coe.c

```diff
@@ -1027,32 +1027,41 @@ static void SDO_downloadsegment (void)
 
       if (coesdo->command & COE_COMMAND_LASTSEGMENTBIT)
       {
-         /* last segment */
-         ESCvar.segmented = 0;
-
          if(ESCvar.flags == COMPLETE_ACCESS_FLAG)
          {
-            int16_t nidx;
+            int16_t nidx, nsub;
+
+            if(ESCvar.frags > ESCvar.fragsleft + size)
+            {
+               set_state_idle (0, ESCvar.index, ESCvar.subindex, ABORT_TYPEMISMATCH);
```

Comment:
A return statement is missing after the call to set_state_idle().

Reply:
fixed